processor/spanmetricsprocessor/processor.go

```diff
@@ -237,6 +237,15 @@ func (p *processorImp) buildMetrics() *pdata.Metrics {
 	p.lock.RLock()
 	p.collectCallMetrics(ilm)
 	p.collectLatencyMetrics(ilm)
+
+	// If delta metrics, flush accumulated data
+	if p.config.GetAggregationTemporality() == pdata.MetricAggregationTemporalityDelta {
```

Comment:
This is not a safe operation to do while obtaining the read lock.

This should use `p.lock.Lock()` and `p.lock.Unlock()` respectively

Reply:
This is a good catch. I've changed it to a write lock.

I've also realised another possible race condition, previously `p.resetExemplarData()` was called outside any lock so I've put it inside the new write lock. It slightly brings forward the function call, so it is called before exporting, but from looking at the code that change any behaviour. 

cc @anneelisabethlelievre I noticed that you worked on the resetting of exemplars recently, let me know if you have any thoughts.